web/packages/teleport/src/Console/DocumentDb/ConnectDialog.tsx

```diff
@@ -0,0 +1,209 @@
+  );
+
+  useEffect(() => {
+    void getDatabase();
+  }, [getDatabase]);
+
+  return (
+    <Dialog
+      dialogCss={dialogCss}
+      disableEscapeKeyDown={false}
+      onClose={props.onClose}
+      open={true}
+    >
+      <DialogHeader mb={4}>
+        <DialogTitle>Connect To Database</DialogTitle>
+      </DialogHeader>
+
+      {attempt.status === 'error' && <Danger children={attempt.statusText} />}
+      {(attempt.status === '' || attempt.status === 'processing') && (
+        <Box textAlign="center" m={10}>
+          <Indicator />
+        </Box>
+      )}
+      {attempt.status === 'success' && (
+        <ConnectForm
+          db={attempt.data}
+          onConnect={props.onConnect}
+          onClose={props.onClose}
+        />
+      )}
+    </Dialog>
+  );
+}
+
+function ConnectForm(props: {
+  db: Database;
+  onConnect(data: DbConnectData): void;
+  onClose(): void;
+}) {
+  const dbUserOpts = props.db.users?.map(user => ({
+    value: user,
+    label: user,
+  }));
+  const dbNamesOpts = props.db.names?.map(name => ({
+    value: name,
+    label: name,
+  }));
+  const dbRolesOpts = props.db.roles?.map(role => ({
+    value: role,
+    label: role,
+  }));
+
+  const [selectedName, setSelectedName] = useState<Option>(dbNamesOpts?.[0]);
+  const [selectedUser, setSelectedUser] = useState<Option>(dbUserOpts?.[0]);
+  const [selectedRoles, setSelectedRoles] = useState<readonly Option[]>();
+
+  const dbConnect = () => {
+    props.onConnect({
+      serviceName: props.db.name,
+      protocol: props.db.protocol,
+      dbName: selectedName.value,
+      dbUser: selectedUser.value,
+      dbRoles: selectedRoles?.map(role => role.value),
+    });
+  };
+
+  return (
+    <Validation>
+      {({ validator }) => (
+        <form>
+          <DialogContent minHeight="240px" flex="0 0 auto">
+            <FieldSelectCreatable
+              label="Database user"
+              menuPosition="fixed"
+              onChange={option => setSelectedUser(option as Option)}
+              value={selectedUser}
+              options={dbUserOpts}
+              isDisabled={dbUserOpts?.length == 1}
+              formatCreateLabel={userInput =>
+                `Use "${userInput}" database user`
+              }
+              rule={requiredField('Database user is required')}
+            />
+            {dbRolesOpts?.length > 0 && (
+              <FieldSelect
+                label="Database roles"
+                menuPosition="fixed"
+                isMulti={true}
+                onChange={setSelectedRoles}
+                value={selectedRoles}
+                options={dbRolesOpts}
+              />
+            )}
+            <FieldSelectCreatable
+              label="Database name"
+              menuPosition="fixed"
+              onChange={option => setSelectedName(option as Option)}
+              value={selectedName}
+              options={dbNamesOpts}
+              isDisabled={dbNamesOpts?.length == 1}
```

Comment:
We might better have a specific field for this decision (show/hide the field). For now, the lack of information, we can use this check. It might only be accurate for some databases, but it works for what we have now.

Reply:
we have this logic already in frontend code, we can refactor this later to a better place and use it here:
https://github.com/gravitational/teleport/blob/57e29f404097e7000e242e06ee35ce29dc15531f/web/packages/teleport/src/Databases/ConnectDialog/ConnectDialog.tsx#L48-L73